pkg/storage/s3/s3.go

```diff
@@ -45,23 +52,28 @@ type driver struct {
 
 	// roundTripper is used only during tests.
 	roundTripper http.RoundTripper
+
+	// used during tests to insert custom credentials
+	GetCloudCredentials func(*regopclient.Listers) (string, error)
```

Comment:
btw, do we need it? We mock listers to provide custom data.

Reply:
:+1:  It was a bit of a mess to use the listers because the in-cluster pieces still need working AWS configs (which look different than the out-of-cluster AWS clients), so I modified it so that the e2e builds it's own AWS client and doesn't rely on running the S3 driver locally to query the state of things.